python/interpret_community/dataset/dataset_wrapper.py

```diff
@@ -19,6 +21,50 @@
     from shap.common import DenseData
 
 
+class CustomTimestampFeaturizer(BaseEstimator, TransformerMixin):
+    def __init__(self, features):
+        self._features = features
+        return
+
+    def fit(self, X):
+        # If the data was previously successfully summarized, then there are no
+        # timestamp columns as it must be numeric.
+        # Also, if the dataset is sparse, we can assume there are no timestamps
+        if isinstance(X, DenseData) or sp.sparse.issparse(X):
+            return self
+        tmp_dataset = X
+        # Temporarily convert to pandas for easier and uniform timestamp handling
+        if isinstance(X, np.ndarray):
+            tmp_dataset = pd.DataFrame(X, columns=self._features)
+        self._time_col_names = [column for column in tmp_dataset.columns if is_datetime(tmp_dataset[column])]
+        # Calculate the min date for each column
+        self._min = []
+        if self._time_col_names:
+            for time_col_name in self._time_col_names:
+                self._min.append(tmp_dataset[time_col_name].map(lambda x: x.timestamp()).min())
+        return self
+
+    def transform(self, X):
+        tmp_dataset = X
+        # Temporarily convert to pandas for easier and uniform timestamp handling
+        if isinstance(X, np.ndarray):
+            tmp_dataset = pd.DataFrame(X, columns=self._features)
+        if self._time_col_names:
+            # Get the year, day, month, hour, minute, second
+            for idx, time_col_name in enumerate(self._time_col_names):
+                tmp_dataset[time_col_name + "_year"] = tmp_dataset[time_col_name].map(lambda x: x.year)
+                tmp_dataset[time_col_name + "_month"] = tmp_dataset[time_col_name].map(lambda x: x.month)
+                tmp_dataset[time_col_name + "_day"] = tmp_dataset[time_col_name].map(lambda x: x.day)
+                tmp_dataset[time_col_name + "_hour"] = tmp_dataset[time_col_name].map(lambda x: x.hour)
+                tmp_dataset[time_col_name + "_minute"] = tmp_dataset[time_col_name].map(lambda x: x.minute)
+                tmp_dataset[time_col_name + "_second"] = tmp_dataset[time_col_name].map(lambda x: x.second)
+                # Replace column itself with difference from min value, leave as same name
+                # to keep index so order of other columns remains the same for other transformations
+                tmp_dataset[time_col_name] = tmp_dataset[time_col_name].map(lambda x: x.timestamp() - self._min[idx])
+            X = tmp_dataset.values
```

Comment:
Do we want to return numpy if X is sparse or pandas?

Reply:
discussed offline; not dealing with sparse case, just returning in fit currently